codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2Connection.java

```diff
@@ -833,7 +832,7 @@ public void flowController(F flowController) {
         }
 
         private void checkNewStreamAllowed(int streamId) throws Http2Exception {
-            if (isGoAway()) {
+            if (goAwaySent() || goAwayReceived()) {
```

Comment:
@nmittler +1 for removing `isGoAway()`. I think this method leads to more harm than good. Having to actively think about we want to check for whether the GOAWAY was sent or received is a lot better.